global/scripts/parallel.shlib

```diff
@@ -172,3 +178,16 @@ function par_addjob_stdin()
     par_jobinputarray+=("$jobinput")
 }
 
+#wait for jobs submitted with fslsub, to finish.
+function par_waitjobs_fslsub()
+{
+    local any_running=1 jobs="$@"
+    while (( any_running )); do
+        any_running=0
+        for job in $jobs; do
+            qstat -j "$job" &> /dev/null
+            if [ $? -eq 0 ]; then any_running=1; break; fi
```

Comment:
That doesn't work under `debug.shlib` and `set -euE`, anything that has nonzero exit will immediately call the `debug.shlib` trap, and you will never reach this line.
```suggestion
            if qstat -j "$job" &> /dev/null; then
                any_running=1
                break
            fi
```